package-res/resources/web/pentaho/type/instance.js

```diff
@@ -0,0 +1,274 @@
+     * @class
+     * @abstract
+     * @amd {pentaho.type.Factory<pentaho.type.Instance>} pentaho/type/instance
+     *
+     * @classDesc The base **instance class** of types in the Pentaho Client Metadata Model.
+     *
+     * Types of the metadata model are constituted by two classes (or constructors):
+     * the **instance class** and the **type class**.
+     *
+     * The former creates the actual instances of the type.
+     * The latter creates a single object (a singleton) that **represents the type**,
+     * is shared by all instances of the type,
+     * and, essentially, holds its metadata.
+     *
+     * The **type class** of the _Instance_ type is {@link pentaho.type.Type}.
+     *
+     * When creating a subclass `Foo` of [Instance]{@link pentaho.type.Instance},
+     * the corresponding type class is implicitly generated
+     * (a subclass of [Type]{@link pentaho.type.Type}),
+     * and its singleton object is placed in the static property
+     * [Foo.type]{@link pentaho.type.Instance.type}.
+     *
+     * Instances of the type `Foo` can also conveniently access the type's singleton object
+     * through the instance property [this.type]{@link pentaho.type.Instance#type}.
+     *
+     * The instance and type classes of a type are closely bound and
+     * must naturally reference each other.
+     * The type singleton object references back the prototype of the instance class,
+     * in a property named [instance]{@link pentaho.type.Type#instance}.
+     *
+     * @example
+     * <caption>
+     *   Create a new class <code>Derived</code> containing an attribute
+     *   <code>greeting</code> and a method <code>doSomething</code>.
+     * </caption>
+     *
+     * require(["pentaho/type/Context"], function(Context) {
+     *   var context = new Context();
+     *   var Instance = context.get("instance");
+     *
+     *   var Derived = Instance.extend({
+     *     constructor: function(label) {
+     *       this.label = label;
+     *     },
+     *     type: { // type specification
+     *       greeting: "Hello, ",
+     *       veryLongString: "..."
+     *     },
+     *     saySomething: function() {
+     *       console.log(this.type.greeting + this.label + "!");
+     *     }
+     *   });
+     *
+     *   var a = new Derived("Alice");
+     *   a.saySomething(); // "Hello, Alice!"
+     *
+     *   var b = new Derived("Bob");
+     *   b.saySomething(); // "Hello, Bob!"
+     *
+     *   // All instances share the same _type_:
+     *   b.type.greeting ===  a.type.greeting // true
+     * });
+     *
+     * @description Creates an instance of this type.
+     *
+     * @see pentaho.type.spec.IInstance
+     * @see pentaho.type.spec.IInstanceProto
+     * @see pentaho.type.spec.ITypeProto
+     */
+    var Instance = Base.extend("pentaho.type.Instance", /** @lends pentaho.type.Instance# */{
+      // NOTE: not calling base to block default Base.js from copying 1st argument into `this`.
```

Comment:
Suggest:

NOTE: Base is not called to block default Base.js from copying 1st argument into `this`.